tests/clock/test_clock.py

```diff
@@ -0,0 +1,427 @@
+    pytest.mark.clock
+]
+
+
+@contextmanager
+def allure_step(step_msg):
+    """
+    @summary:
+        Context manager that wraps allure step context and a log with the same message
+    @param step_msg: The desired step message
+    """
+    with allure.step(step_msg) as allure_step_context:
+        logging.info('Step start: {}'.format(step_msg))
+        try:
+            yield allure_step_context
+        finally:
+            logging.info('Step end: {}'.format(step_msg))
+
+
+class ClockConsts:
+    STDOUT = "stdout"
+    STDERR = "stderr"
+
+    DATE = "date"
+    TIME = "time"
+    TIMEZONE = "timezone"
+
+    TEST_TIMEZONE = "Asia/Jerusalem"
+    TIME_MARGIN = 6
+    RANDOM_NUM = 6
+
+    # sonic commands
+    CMD_SHOW_CLOCK = "show clock"
+    CMD_SHOW_CLOCK_TIMEZONES = "show clock timezones"
+    CMD_CONFIG_CLOCK_TIMEZONE = "config clock timezone"
+    CMD_CONFIG_CLOCK_DATE = "config clock date"
+
+    # expected outputs
+    OUTPUT_CMD_SUCCESS = ''
+
+    # expected errors
+    ERR_BAD_TIMEZONE = 'Timezone {} does not conform format'
+    ERR_MISSING_DATE = 'Error: Missing argument "<YYYY-MM-DD>"'
+    ERR_MISSING_TIME = 'Error: Missing argument "<HH:MM:SS>"'
+    ERR_BAD_DATE = 'Date {} does not conform format YYYY-MM-DD'
+    ERR_BAD_TIME = 'Time {} does not conform format HH:MM:SS'
+
+    # timedatectl
+    CMD_TIMEDATECTL = "timedatectl"
+    TIME_ZONE = "Time zone"
+
+    MIN_SYSTEM_DATE = "1970-01-01"
+    MAX_SYSTEM_DATE = "2231-12-31"
+
+    # ntp
+    CMD_SHOW_NTP = "show ntp"
+    CMD_CONFIG_NTP_ADD = "config ntp add"
+    CMD_CONFIG_NTP_DEL = "config ntp del"
+    OUTPUT_CMD_NTP_ADD_SUCCESS = 'NTP server {} added to configuration\nRestarting ntp-config service...'
+    OUTPUT_CMD_NTP_DEL_SUCCESS = 'NTP server {} removed from configuration\nRestarting ntp-config service...'
+    REGEX_NTP_POLLING_TIME = r'polling server every (\d+)'
+
+
+class ClockUtils:
+    @staticmethod
+    def run_cmd(duthosts, cmd, param=''):
+        """
+        @summary:
+            Run a given command and return its output.
+            * A successful command returns an empty output (except for show commands),
+                while failure returns an error message
+        @return: commands output (str)
+        """
+        with allure_step('Run command: "{}" with param "{}"'.format(cmd, param)):
+            dut_hostname = duthosts[0].hostname
+
+            cmd_to_run = cmd if param == '' else cmd + ' ' + param
+            logging.info('Actual command to run: "{}"'.format(cmd_to_run))
+
+            try:
+                cmd_output = duthosts.command(cmd_to_run)[dut_hostname][ClockConsts.STDOUT]
+            except RunAnsibleModuleFail as cmd_err:
+                output = cmd_err.results[ClockConsts.STDOUT]
+                err = cmd_err.results[ClockConsts.STDERR]
+                cmd_output = output if output else err
+                logging.info('Command Error!\nError message: "{}"'.format(cmd_output))
```

Comment:
should this be defined as error or info?

Reply:
i think info, because its not an error with the test, its an error message which may be expected, so just print it to log to be noted. agree, or should change to error? @roy-sror

Reply:
Agree